pkg/apis/monitoring/v1/types.go

```diff
@@ -426,6 +432,25 @@ type PrometheusRuleExcludeConfig struct {
 	RuleName string `json:"ruleName"`
 }
 
+// ObjectReference references a PodMonitor, ServiceMonitor, Probe or PrometheusRule object.
+type ObjectReference struct {
+	// Group of the referent.
+	Group string `json:"group"`
+	// Resource of the referent.
+	// +kubebuilder:validation:Enum=prometheusrules;servicemonitors;podmonitors;probes
+	Resource string `json:"resource"`
+	// Namespace of the referent.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	Namespace string `json:"namespace"`
```

Comment:
Sorry about that but my earlier missed to mention that the `name` of the referent was needed.

```suggestion
	Namespace string `json:"namespace"`
	// Name of the referent.
	Name string `json:"name"`
```

Reply:
:) - would that `name` be optional?
My use-case would still be (at minimum) to  `avoid enforcement for all monitoring resources in a given namespace`

Reply:
yes i'm ok to consider that an empty `name` means no enforcement for all objects of the given resource in the given namespace.

Reply:
@simonpasquier Added `Name` field to the reference object.